src/javascript/app_2/Stores/Modules/Statement/statement_store.js

```diff
@@ -84,22 +99,18 @@ export default class StatementStore extends BaseStore {
 
     @action.bound
     onMount() {
+        eventBus.listen('ClientAccountHasSwitched', () => {
```

Comment:
@easteregg 
is it not better to use [`when`](https://mobx.js.org/refguide/when.html) or [`reaction`](https://mobx.js.org/refguide/reaction.html) instead?

Reply:
Would you elaborate on what you mean by that? using reaction makes these 2 stores to be more coupled together, (let's say we create a "flag" to activate when change happens to client), our statement store and trade store and everything else (that need to do something is going to be reliant on that particular `variable` from the store). Is this going to be a better approach, instead of doing it the pub/sub way?

Reply:
by using pub/sub event, those stores are imposed to depend on client_store too. 
BTW the client_store is the store that other stores are based on it since they need login client information. 
IMO, using mobx facilities is more reliable to handle the flow of changes.